lib/ansible/plugins/inventory/netbox.py

```diff
@@ -242,7 +242,12 @@ def extract_tenant(self, host):
 
     def extract_device_role(self, host):
         try:
-            return [self.device_roles_lookup[host["device_role"]["id"]]]
+            if "device_role" in host:
+                return [self.device_roles_lookup[host["device_role"]["id"]]]
+            elif "role" in host:
+                return [self.device_roles_lookup[host["role"]["id"]]]
+            else:
+                return
```

Comment:
```
else:
     return
```

is useless

Reply:
I've just pushed a new commit without the two lines.